IdentityCore/src/MSIDDeviceId.h

```diff
@@ -35,6 +35,15 @@
 /*! Returns a short device identifier string containing device type and OS version. */
 + (NSString *)deviceOSId;
 
+/*! Returns a unique device identifier for telemetry purposes. */
++ (NSString *)deviceTelemetryId;
+
+/*! Returns application name for telemetry purposes. */
++ (NSString *)applicationName;
+
+/*! Returns application version for telemetry purposes. */
++ (NSString *)applicationVersion;
```

Comment:
nit: Since we return not only "ids" but more general device info like "applicationVersion" and "applicationName", should we consider to rename the class from "MSIDDeviceId" to "MSIDDevice" ?

Reply:
Agree, I was thinking the same. I'll do it in a separate PR, created an issue: https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/issues/17